core/io/file_access_pack.cpp

```diff
@@ -123,16 +123,19 @@ PackedData::~PackedData() {
 
 //////////////////////////////////////////////////////////////////
 
-bool PackedSourcePCK::try_open_pack(const String &p_path, bool p_replace_files) {
+bool PackedSourcePCK::try_open_pack(const String &p_path, bool p_replace_files, size_t p_offset) {
 	FileAccess *f = FileAccess::open(p_path, FileAccess::READ);
 	if (!f) {
 		return false;
 	}
 
+	f->seek(p_offset);
+
 	uint32_t magic = f->get_32();
 
-	if (magic != PACK_HEADER_MAGIC) {
+	if (magic != PACK_HEADER_MAGIC && p_offset == 0) {
```

Comment:
It seems like the magic isn't actually checked anymore once the offset is not equal to zero.  There should be a check to make sure the magic is valid at the new offset in order to help avoid loading random data as if it was a PCK, no?

Reply:
I think that you are correct. In current implementation this if block will be skipped if offset is none zero even though PACK_HEADER_MAGIC don't match. I updated this line and moved the ```p_offset == 0``` check inside this if statement block.

Reply:
LGTM, thanks.